### PR TITLE
Update description of --solver-plugin

### PR DESCRIPTION
**Pull request by keyboardDrummer**
Updated description reads:

```
  --solver-plugin <solver-plugin>              Dafny uses Boogie as part of its verification process. This 
                                               option allows customising that part using a Boogie plugin. 
                                               More information about Boogie can be found at 
                                               https://github.com/boogie-org/boogie. Information on how to 
                                               construct Boogie plugins can be found by looking at the code 
                                               in 
                                               https://github.com/boogie-org/boogie/blob/58ab09236f41894ee9a7
                                               93d30eb17617b0fce797/Source/Provers/SMTLib/ProverUtil.cs
```

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyCore/Options/CommonOptionBag.cs [hunk 1]
```diff
@@ -61,7 +61,8 @@ public class CommonOptionBag {
   };
 
   public static readonly Option<uint> SolverResourceLimit = new("--resource-limit", @"Specify the maximum resource limit (rlimit) value to pass to Z3. Multiplied by 1000 before sending to Z3.");
-  public static readonly Option<string> SolverPlugin = new("--solver-plugin", @"Specify a plugin to use to solve verification conditions (instead of an external Z3 process).");
+  public static readonly Option<string> SolverPlugin = new("--solver-plugin",
+    @"Dafny uses Boogie as part of its verification process. This option allows customising that part using a Boogie plugin. More information about Boogie can be found at https://github.com/boogie-org/boogie. Information on how to construct Boogie plugins can be found by looking at the code in https://github.com/boogie-org/boogie/blob/58ab09236f41894ee9a793d30eb17617b0fce797/Source/Provers/SMTLib/ProverUtil.cs");
```
Comment (atomb) [inline Source/DafnyCore/Options/CommonOptionBag.cs:65]: It seems to me that that link could get out of date quickly. Could you change it to a link that'd resolve to the latest version, instead?

**Review by atomb — CHANGES_REQUESTED** [review]

File: Source/DafnyCore/Options/CommonOptionBag.cs (hunk 1 above)
Comment (keyboardDrummer) [reply Source/DafnyCore/Options/CommonOptionBag.cs:65]: An 'up-to-date' link will break when the file gets moved. I think a potentially out-of-date but unbreakable link is better. Neither of these two options is perfect. Perfect would be a stable link to documentation but Boogie doesn't have documentation on this topic.

Comment (atomb) [reply Source/DafnyCore/Options/CommonOptionBag.cs:65]: Would it be possible to link to a file as it exists for a certain tag? Then we could keep things pointing at the tag of the most recent release.

Comment (keyboardDrummer) [reply Source/DafnyCore/Options/CommonOptionBag.cs:65]: I'm not sure what you mean. Either you point to a fixed point in time, using a commit SGA or tag, and then you're sure the path doesn't break, if you point to a dynamic point in time, and then the file path may break. Right?

**Review by atomb — APPROVED** [review]
Love it!

**Pull request merged**